Lesson_5/Irrational_numbers/IrrationalNumber.cs

```diff
@@ -0,0 +1,100 @@
+        }
+        public static IrrationalNumber operator *(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return new IrrationalNumber(num1._Numerator * num2._Numerator, num1._Denominator * num2._Denominator);
+        }
+        public static IrrationalNumber operator /(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return new IrrationalNumber(num1._Numerator * num2._Denominator, num1._Denominator * num2._Numerator);
+        }
+        public static IrrationalNumber operator %(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return new IrrationalNumber((num1._Numerator * num2._Denominator) % (num1._Denominator * num2._Numerator), 1);
+        }
+        public static bool operator ==(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) == ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static bool operator !=(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) != ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static bool operator >(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) > ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static bool operator <(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) < ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static bool operator >=(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) >= ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static bool operator <=(IrrationalNumber num1, IrrationalNumber num2)
+        {
+            return (((double)num1._Numerator / (double)num1._Denominator) <= ((double)num2._Numerator / (double)num2._Denominator));
+        }
+        public static IrrationalNumber operator ++(IrrationalNumber num1)
+        {
+            return new IrrationalNumber(num1._Numerator + num1._Denominator, num1._Denominator);
+        }
+        public static IrrationalNumber operator --(IrrationalNumber num1)
+        {
+            if (num1._Numerator > num1._Denominator)
+                return new IrrationalNumber(num1._Numerator - num1._Denominator, num1._Denominator);
+            return new IrrationalNumber(-(num1._Denominator - num1._Numerator), num1._Denominator);
+        }
+        public static implicit operator float(IrrationalNumber num1)
+        {
+            return (float)num1._Numerator / (float)num1._Denominator;
+        }
```

Comment:
```suggestion
        public static implicit operator double(IrrationalNumber num1)
        {
            return (double)num1._Numerator / (double)num1._Denominator;
        }
```
У float слишком низкая точность.